core/src/data/tileSource.cpp

```diff
@@ -122,9 +115,12 @@ void TileSource::cancelLoadingTile(TileTask& _task) {
 }
 
 void TileSource::addRasterSource(std::shared_ptr<TileSource> _rasterSource) {
-    /*
-     * We limit the parent source by any attached raster source's min/max.
-     */
+    auto rasterSource = dynamic_cast<RasterSource*>(_rasterSource.get());
+    if (!rasterSource) {
+        assert(false);
```

Comment:
This seems like it would be triggered if a user assigns a data source of the wrong type to a `rasters` property in the scene file. We shouldn't crash the app on scene file errors!